File: src/MiningMonitor.Model/AlertDefinition.cs
using System;
using System.Collections.Generic;

using LiteDB;

namespace MiningMonitor.Model
{
    public class AlertDefinition
    {
        private Dictionary<string, string> _parameters = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);

        [BsonId(false)]
        public Guid Id { get; set; }
        public Guid MinerId { get; set; }
        public AlertType AlertType { get; set; }
        public bool Enabled { get; set; }
        public DateTime Created { get; set; }
        public DateTime? Updated { get; set; }
        public DateTime? LastEnabled { get; set; }
        public DateTime? LastTest { get; set; }

        public Dictionary<string, string> Parameters
        {
            get => _parameters;
            set => _parameters = new Dictionary<string, string>(value, StringComparer.OrdinalIgnoreCase);
        }
    }
}

File: src/MiningMonitor.Web/ClientApp/src/app/models/AlertDefinition.ts
import { AlertActionDefinition } from './AlertActionDefinition';
import { AlertParameters } from './AlertParameters';
import { AlertSeverity } from './AlertSeverity';

export interface AlertDefinition {
  id?: string;
  minerId: string;
  severity?: AlertSeverity;
  displayName: string;
  enabled: boolean;
  parameters: AlertParameters;
  actions: AlertActionDefinition[];
  created?: Date;
  updated?: Date;
  lastEnabled?: Date;
  lastScan?: Date;
  name?: string;
}
